qiling/extensions/r2/r2.py

```diff
@@ -12,21 +12,27 @@
 from qiling.core import Qiling
 
 @dataclass(unsafe_hash=True)
-class Function:
+class R2Data:
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass(unsafe_hash=True)
+class Function(R2Data):
     name: str
     offset: int
     size: int
     signature: str
 
     def __init__(self, **kwargs):
-        names = set([f.name for f in fields(self)])
-        for k, v in kwargs.items():
-            if k in names:
-                setattr(self, k, v)
+        super().__init__(**kwargs)
```

Comment:
When all what the `__init__` method does is to call its `super` `__init__` method with no special processing, you could just omit it altogether. In other words: the subclasses `__init__` methods can be removed.

Reply:
At first I only declare `__init__()` on the base class, but the fields of subclasses will not work. I don't know if there is anything special with dataclass.